marketplace/google-analytics-app/src/AppConfig.tsx

```diff
@@ -0,0 +1,342 @@
+      >,
+      getAndUpdateSavedParams(sdk)
+    ]);
+
+    const { contentTypes } = savedParams;
+
+    // add an incomplete contentType entry if there are none saved
+    if (!Object.keys(contentTypes).length) {
+      contentTypes[''] = { slugField: '', urlPrefix: '' };
+    }
+
+    // eslint-disable-next-line react/no-did-mount-set-state
+    this.setState(
+      {
+        // sort contentTypes by display name
+        allContentTypes: sortBy(spaceContentTypes, 'name').reduce(
+          (acc: AppConfigState['allContentTypes'], contentType) => {
+            acc[contentType.sys.id] = {
+              ...contentType,
+              fields: sortBy(
+                // use only short text fields of content type
+                contentType.fields.filter(f => f.type === 'Symbol'),
+                // sort by field name
+                'name'
+              )
+            };
+
+            return acc;
+          },
+          {}
+        ),
+        contentTypes,
+        clientId: savedParams.clientId || '',
+        viewId: savedParams.viewId || ''
+      },
+      () => sdk.app.setReady()
+    );
+
+    sdk.app.onConfigure(() => this.configureApp());
+  }
+
+  async configureApp() {
+    const { contentTypes, clientId, viewId } = this.state;
+    const { notifier } = this.props.sdk;
+
+    if (!clientId || !viewId) {
+      notifier.error('You must provide both a valid client ID and view ID!');
+
+      return false;
+    }
+
+    if (!/^[-a-z0-9]+.apps.googleusercontent.com$/i.test(clientId)) {
+      notifier.error("The value given for the client ID doesn't look valid!");
+      return false;
+    }
+
+    const ctKeys = Object.keys(contentTypes);
+
+    if (!ctKeys.length || !ctKeys[0]) {
+      notifier.error('You need to select at least one content type with a slug field!');
+    }
+
+    if (ctKeys.some(key => key && !contentTypes[key].slugField)) {
+      notifier.error('Please complete or remove the incomplete content type rows!');
+      return false;
+    }
+
+    const editorInterface = ctKeys.reduce(
+      (acc: { [key: string]: Partial<EditorInterface> }, id) => {
+        const sidebarPosition: { [key: string]: object } = { sidebar: { position: 1 } };
+
+        acc[id] = sidebarPosition;
+        return acc;
+      },
+      {}
+    );
+
+    return {
+      parameters: {
+        contentTypes,
+        clientId,
+        viewId
+      },
+      targetState: {
+        EditorInterface: editorInterface
+      }
+    };
+  }
+
+  handleContentTypeChange(prevKey: string, newKey: string) {
+    this.setState(prevState => {
+      const contentTypes: AppConfigState['contentTypes'] = {};
+
+      // remove contentType[prevKey] field and replace with the new contentType
+      // key while preserving key order
+      for (const [prop, value] of Object.entries(prevState.contentTypes)) {
+        if (prop === prevKey) {
+          contentTypes[newKey as keyof typeof contentTypes] = {
```

Comment:
won't it simply accept `newKey` as a string?

Reply:
No, because `contentTypes` has no `[key: string]: SpecificType` decleration, so `contentTypes[string]` is `any`.